walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,113 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.HashSet;
+
+/**
+* The FindMeetingQuery receives a new Meeting request to fit inside a
+* calendar day with existing meetings. It provides an interface to
+* handle incoming meetings that must be scheduled for users without
+* generating interference/overlapping with their existing meetings.
+* 
+* Algorithmic analysis
+* N = Scheduled events
+* M = Attendees within scheduled events
```

Comment:
I think M is worst case "the entire pool of people who can attend events", because of Collections.disjoint() iterating over both input sets, and being called for every event.